### PR TITLE
Free buffers upon exit.

### PR DESCRIPTION
**Pull request by gameblabla**
Should fix leaking memory upon exit.

### CODE REVIEW

File: src/video/tinspire/SDL_tinspirevideo.c [hunk 1]
```diff
@@ -288,6 +288,17 @@ static int NSP_SetColors(_THIS, int firstcolor, int ncolors, SDL_Color *colors)
 */
 static void NSP_VideoQuit(_THIS)
 {
+	if (this->hidden->buffer != NULL)
```
Comment (Vogtinator) [inline src/video/tinspire/SDL_tinspirevideo.c:291]: `free(NULL);`  is defined to be a noop, so the `if` can be removed

Comment (gameblabla) [reply src/video/tinspire/SDL_tinspirevideo.c:291]: So i should just free buffer2 ?

Comment (gameblabla) [reply src/video/tinspire/SDL_tinspirevideo.c]: Alright i did it the proper way but not sure what you meant by removing the if statement. I think you meant that i should have removed the != NULL instead.

File: src/video/tinspire/SDL_tinspirevideo.c [hunk 2]
```diff
@@ -288,6 +288,12 @@ static int NSP_SetColors(_THIS, int firstcolor, int ncolors, SDL_Color *colors)
 */
 static void NSP_VideoQuit(_THIS)
 {
+	if ( this->hidden->buffer ) {
```
Comment (Vogtinator) [inline src/video/tinspire/SDL_tinspirevideo.c:291]: What I meant is that you can do just this, unconditionally:

```
SDL_free(this->hidden->buffer);
this->hidden->buffer = NULL;
SDL_free(this->hidden->buffer2);
this->hidden->buffer2 = NULL;
```

Comment (gameblabla) [reply src/video/tinspire/SDL_tinspirevideo.c:291]: Oh i see... Strange that it's not done this way when cleaning the buffer upon allocating it but i'll do just that.

**Pull request merged**